src/mixins/widget.js

```diff
@@ -22,6 +22,11 @@ export const createWidgetMixin = ({ connector } = {}) => ({
       this.factory = connector(this.updateState, () => {});
       this.widget = this.factory(this.widgetParams);
       this.instantSearchInstance.addWidget(this.widget);
+
+      const { hydrated, started } = this.instantSearchInstance;
+      if ((!started && hydrated) || this.$isServer) {
```

Comment:
We don't have the blink anymore but we still call `forceRender` even when we don't have to. The `hydrated` value is never reset to `false`. So even without a value we pay the cost of an extra render. We can check the value inside the helper or (to avoid to rely on implementation detail) sync the `hydrated` value with the helper.